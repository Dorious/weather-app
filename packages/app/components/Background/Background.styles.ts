import styled from 'styled-components'

import { AbsoluteDiv } from '../styles'

export type StyledBackgroundProps = {
  sky: string
  daytime: string
}

export const Background = styled(AbsoluteDiv)<StyledBackgroundProps>`
  background: #d1d1d1;
  z-index: -1;
  position: fixed;
`
