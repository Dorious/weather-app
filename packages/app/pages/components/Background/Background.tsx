import * as S from './Background.styles';

export type BackgroundProps = {
  symbolCode: string;
};

export default function Background({symbolCode}: BackgroundProps): JSX.Element {
  const [sky, daytime] = symbolCode.split('_');

  return (
    <S.Background sky={sky} daytime={daytime} />
  );
};