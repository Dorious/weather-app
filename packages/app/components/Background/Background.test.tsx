import { render, screen } from '@testing-library/react';
import Background from '.';

describe('SearchBar', () => {
  it('should render input', () => {
    const { container } = render(<Background symbolCode="clearsky_day" />);
    expect(container).toMatchSnapshot();
  });
});