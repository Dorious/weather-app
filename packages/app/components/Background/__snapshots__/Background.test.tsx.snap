// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render input 1`] = `
<div>
  <div
    class="styles__AbsoluteDiv-i2fsqp-0 Backgroundstyles__Background-sc-1qaidi-0 bjyhA hTdLPQ"
  />
</div>
`;
